sn_client/src/wallet.rs

```diff
@@ -104,86 +103,103 @@ impl WalletClient {
     /// This can optionally verify the store has been successful (this will attempt to GET the dbc from the network)
     pub async fn pay_for_storage(
         &mut self,
-        content_addrs: impl Iterator<Item = &XorName>,
+        content_addrs: impl Iterator<Item = NetworkAddress>,
         verify_store: bool,
-    ) -> Result<(PaymentProofsMap, Option<Token>)> {
-        // Let's filter the content addresses we hold payment proofs for, i.e. avoid
-        // paying for those chunks we've already paid for with this wallet.
-        let mut proofs = PaymentProofsMap::default();
-
-        let addrs_to_pay: Vec<&XorName> = content_addrs.collect();
-        // TODO: reenable this when we have a way to get the store cost from the network
-        // per chunk, and can readily check what we've paid here.
-        // .filter(|name| {
-        //     if let Some(proof) = self.wallet.get_payment_proof(name) {
-        //         proofs.insert(**name, proof.clone());
-        //         false
-        //     } else {
-        //         true
-        //     }
-        // })
-        // .collect();
-
-        let number_of_records_to_pay = addrs_to_pay.len() as u64;
-
-        // If no addresses need to be paid for, we don't have to go further
-        if addrs_to_pay.is_empty() {
-            trace!("We already hold payment proofs for all the records.");
-            return Ok((proofs, None));
+    ) -> Result<(TransferOutputsMap, Token)> {
+        let mut total_cost = Token::zero();
+
+        let mut payment_map = BTreeMap::default();
+
+        let mut tasks = JoinSet::new();
+        // we can collate all the payments together into one transfer
```

Comment:
What do we mean by one transfer here? We do 1 transfer per record right?

Reply:
yeh, one per record is what we have right now. That could be improved upon I think